joschahenningsen/TUM-Live
# Data Store

### Motivation and Context
Loading data from the API and storing it was very entangled with the controllers for the UI. This makes it harder if different components need to load the same data resulting in loading sections multiple times. This could also cause inconsistencies.

### Description
A global store that stores the most up-to-date data, and can be subscribed to by different components, would solve the issues. This keeps the current data state consistent across all components that are using it. Also using a customizable cache that minimizes fetches.

### Steps for Testing
Prerequisites:
- 1 Admin
- 1 Livestream

1. Log in
2. Navigate to a Course Admin panel
3. Check if section editing works as expected
4. Go to a live stream that has sections
5. See if sections are displayed correctly
6. Add / Edit / Remove Bookmarks and chek if UI responds correctly

### Screenshots
![Bildschirmfoto 2023-03-30 um 00 27 01](https://user-images.githubusercontent.com/17506411/228681705-084ba3af-c637-4b40-8005-2d0fc6bbab34.png)
